### PR TITLE
got all but 4 tests passing for py3!

### PR DESCRIPTION
**Pull request by parrt**

### CODE REVIEW

File: runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/LeftRecursion/JavaExpressions.stg [hunk 1]
```diff
@@ -17,18 +17,18 @@ expressionList
     :   e (',' e)*
     ;
 e   :   '(' e ')'
-    |   'this' 
+    |   'this'
     |   'super'
     |   INT
     |   ID
-    |   type '.' 'class'
+    |   typespec '.' 'class'
```
Comment (sharwell) [inline runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/LeftRecursion/JavaExpressions.stg]: :exclamation: A new test needs to be added to ensure `type` remains a valid rule name in the Java target.

File: runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/LeftRecursion/MultipleAlternativesWithCommonLabel.stg [hunk 2]
```diff
@@ -19,16 +19,16 @@ Rule() ::= "s"
...
-s : e {<writeln("$e.v")>}; 
+s : e {<writeln("$e.v")>};
 e returns [int v]
   : e '*' e     {$v = <Cast("BinaryContext","$ctx")>.e(0).v * <Cast("BinaryContext","$ctx")>.e(1).v;}  # binary
   | e '+' e     {$v = <Cast("BinaryContext","$ctx")>.e(0).v + <Cast("BinaryContext","$ctx")>.e(1).v;}  # binary
   | INT         {$v = $INT.int;}                   # anInt
   | '(' e ')'   {$v = $e.v;}                       # parens
-  | left=e INC  {<Cast("UnaryContext","$ctx"):Concat(".INC() != null"):Assert()> $v = $left.v + 1;}      # unary
-  | left=e DEC  {<Cast("UnaryContext","$ctx"):Concat(".DEC() != null"):Assert()> $v = $left.v - 1;}      # unary
+  | left=e INC  {<Cast("UnaryContext","$ctx"):Concat(".INC() != null"):Assert()>$v = $left.v + 1;}      # unary
```
Comment (sharwell) [inline runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/LeftRecursion/MultipleAlternativesWithCommonLabel.stg]: :question: Why did this change?

File: runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/ParserExec/PredicatedIfIfElse.stg [hunk 3]
```diff
@@ -25,7 +25,7 @@ grammar(grammarName) ::= <<
 grammar <grammarName>;
 s : stmt EOF ;
 stmt : ifStmt | ID;
-ifStmt : 'if' ID stmt ('else' stmt | { <LANotEquals("1", "ELSE")> }?);
+ifStmt : 'if' ID stmt ('else' stmt | { <LANotEquals("1", {<grammarName>Parser.ELSE})> }?);
```
Comment (sharwell) [inline runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/ParserExec/PredicatedIfIfElse.stg]: :exclamation: A test needs to be added to ensure `ELSE` (unqualified) remains a valid reference in code generated by the Java target.

File: runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/LeftRecursion/MultipleAlternativesWithCommonLabel.stg (hunk 2 above)
Comment (parrt) [inline runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/LeftRecursion/MultipleAlternativesWithCommonLabel.stg]: so python would run. had space; i.e., indentation

File: runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/LeftRecursion/JavaExpressions.stg (hunk 1 above)
Comment (parrt) [inline runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/LeftRecursion/JavaExpressions.stg]: i don't see a need. moving on.

File: runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/ParserExec/PredicatedIfIfElse.stg (hunk 3 above)
Comment (parrt) [inline runtime-testsuite/resources/org/antlr/v4/test/runtime/templates/ParserExec/PredicatedIfIfElse.stg]: not point of test but feel free to add a test.

**Pull request merged**